### PR TITLE
feat(react-tag-picker): adds appearance and size props

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

1. adds `appearance` prop to `TagPicker`
2. adds `size` prop to `TagPicker`

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1db129b125726aca0f500ac1f292eca59e7db57b'>1db129b125726aca0f500ac1f292eca59e7db57b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Avatar-mount_base.html">653</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Avatar-mount.html">622</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Button-mount_base.html">309</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Button-mount.html">303</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Field-mount_base.html">1130</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Field-mount.html">1164</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProvider-mount_base.html">718</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProvider-mount.html">734</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProviderWithTheme-mount_base.html">83</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProviderWithTheme-mount.html">91</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">60</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/MakeStyles-mount_base.html">849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/MakeStyles-mount.html">852</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Persona-mount_base.html">1745</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/Persona-mount.html">1704</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/SpinButton-mount_base.html">1365</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30742/perf-test-react-components/SpinButton-mount.html">1362</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-tag-picker-preview/src/components/TagPicker/useTagPickerContextValues.ts [hunk 1]
```diff
@@ -2,12 +2,64 @@ import * as React from 'react';
 import { TagPickerContextValues, TagPickerState } from './TagPicker.types';
 
 export function useTagPickerContextValues(state: TagPickerState): TagPickerContextValues {
+  const {
+    onOptionClick,
```
Comment (ling1726) [inline packages/react-components/react-tag-picker-preview/src/components/TagPicker/useTagPickerContextValues.ts:6]: I think the unused state/types should be cleaned up in a future PR - not blocking just an FYI

File: packages/react-components/react-tag-picker-preview/src/components/TagPickerControl/useTagPickerControlStyles.styles.ts [hunk 2]
```diff
@@ -156,8 +156,8 @@ export const useTagPickerControlStyles_unstable = (state: TagPickerControlState)
     styles.root,
     styles[state.size],
     styles[state.appearance],
-    !state.disabled && state.appearance === 'outline' && styles.outlineInteractive,
-    state.disabled && styles.disabled,
+    // !state.disabled && state.appearance === 'outline' && styles.outlineInteractive,
+    // state.disabled && styles.disabled,
```
Comment (ling1726) [inline packages/react-components/react-tag-picker-preview/src/components/TagPickerControl/useTagPickerControlStyles.styles.ts:160]: should these stay commented?

File: packages/react-components/react-tag-picker-preview/src/utils/useInputTriggerSlot.ts [hunk 3]
```diff
@@ -83,88 +68,90 @@ export function useInputTriggerSlot(
...
-    defaultOnKeyDown?.(ev);
-  });
-
-  trigger.onKeyDown = onKeyDown;
-
-  if (hideActiveDescendant) {
-    trigger['aria-activedescendant'] = undefined;
-  }
-
+  const trigger = useTriggerSlot(
```
Comment (ling1726) [inline packages/react-components/react-tag-picker-preview/src/utils/useInputTriggerSlot.ts:75]: how much of this has changed? the intention was to eventually re-use the combobox logic

File: packages/react-components/react-tag-picker-preview/src/components/TagPickerControl/useTagPickerControlStyles.styles.ts (hunk 2 above)
Comment (bsunderhus) [reply packages/react-components/react-tag-picker-preview/src/components/TagPickerControl/useTagPickerControlStyles.styles.ts:160]: I'll keep this commented for now as I still haven't investigate all the cases of disabled yet. There are too many controls involved in this component, disabling might not be trivial.

File: packages/react-components/react-tag-picker-preview/src/utils/useInputTriggerSlot.ts (hunk 3 above)
Comment (bsunderhus) [reply packages/react-components/react-tag-picker-preview/src/utils/useInputTriggerSlot.ts:75]: The previous code simply didn't work. It was a merge between the new `activedescendant` logic and an old local one that was basically ignored. I'll see to update `combobox` logic with whatever I change here. The final goal is definitely to have them both equal

**Pull request merged**